sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+        try {
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+
+            return this.registriesImplClient.deleteRepositoryWithResponseAsync(repositoryName, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapDeleteRepositoryResult))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DeleteRepositoryResult> delete() {
+        return this.deleteWithResponse().map(Response::getValue);
+    }
+
+    /**
+     * Delete registry artifact.
+     *
+     * @param digest Digest name.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> deleteRegistryArtifactWithResponse(String digest) {
+        return withContext(context -> this.deleteRegistryArtifactWithResponse(digest, context));
+    }
+
+    Mono<Response<Void>> deleteRegistryArtifactWithResponse(String digest, Context context) {
+        try {
+            if (digest == null) {
+                return monoError(logger, new NullPointerException("'digest' cannot be null"));
+            }
+
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+
+            return this.serviceClient.deleteManifestWithResponseAsync(repositoryName, digest, context)
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Delete registry artifact.
+     *
+     * @param digest digest to delete.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteRegistryArtifact(String digest) {
+        return this.deleteRegistryArtifactWithResponse(digest)
+            .flatMap((Response<Void> res) -> Mono.empty());
```

Comment:
Should still be able to use the `FluxUtil` method to remove the response, correct?

Reply:
Looked at the fluxutil methods and I believe you mean FluxUtil::toMono --> did not know it existed, changed - Thanks!

---
In reply to: [609152718](https://github.com/Azure/azure-sdk-for-java/pull/20368#discussion_r609152718) [](ancestors = 609152718)